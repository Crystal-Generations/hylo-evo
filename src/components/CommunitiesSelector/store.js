import { createSelector } from 'reselect'
import sampleCommunities from './sampleCommunities'

const FIND_COMMUNITIES = 'hyloEditor/FIND_COMMUNITIES'
const CLEAR_COMMUNITIES = 'hyloEditor/CLEAR_COMMUNITIES'

// Action Creators

export function findCommunities (searchText) {
  return {
    type: FIND_COMMUNITIES,
    payload: { searchText }
  }
}

export function clearCommunities (searchText) {
  return { type: CLEAR_COMMUNITIES }
}

// Reducer

const defaultState = {
  communitiesResults: sampleCommunities
}

export default function reducer (state = defaultState, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FIND_COMMUNITIES:
      // filter(i =>
      //   String(i
      //     .get('name')
      //     .replace(/\s+/g, '')
      //     .toLowerCase()
      //     .indexOf(lowerSearch) !== -1
      //   )
      // )
      // .take(5)
      return {...state, communitiesResults: filter(payload.searchText)(sampleCommunities)}
    case CLEAR_COMMUNITIES:
      return {...state, communitiesResults: []}
    default:
      return state
  }
}

// Selectors

export const getCommunitiesResults = createSelector(
  (state) => state.CommunitiesSelector,
  (state, props) => state.communitiesResults
)
