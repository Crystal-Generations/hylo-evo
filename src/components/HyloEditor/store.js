import { createSelector } from 'reselect'
import { fromJS } from 'immutable'
import sampleMentions from './sampleMentions'
import mentionPlugin from 'draft-js-mention-plugin'
import hashtagPlugin from './hashtagPlugin'

const defaultMentionsSuggestionFilter = mentionPlugin.defaultSuggestionsFilter
const defaultHashtagSuggestionFilter = hashtagPlugin.defaultSuggestionsFilter

const FIND_MENTIONS = 'hyloEditor/FIND_MENTIONS'
const CLEAR_MENTIONS = 'hyloEditor/CLEAR_MENTIONS'
const FIND_HASHTAGS = 'hyloEditor/FIND_HASHTAGS'
const CLEAR_HASHTAGS = 'hyloEditor/CLEAR_HASHTAGS'

// Action Creators

export function findMentions (searchText) {
  return {
    type: FIND_MENTIONS,
    payload: { searchText }
  }
}

export function clearMentions (searchText) {
  return { type: CLEAR_MENTIONS }
}

export function findHashtags (searchText) {
  return {
    type: FIND_HASHTAGS,
    payload: { searchText }
  }
}

export function clearHashtags (searchText) {
  return { type: CLEAR_HASHTAGS }
}

// Reducer

export default function reducer (state = {mentionResults: fromJS([])}, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case FIND_MENTIONS:
      return {...state, mentionResults: defaultMentionsSuggestionFilter(payload.searchText, fromJS(sampleMentions))}
    case CLEAR_MENTIONS:
      return {...state, mentionResults: fromJS([])}
    case FIND_HASHTAGS:
      return {...state, mentionResults: defaultHashtagSuggestionFilter(payload.searchText, fromJS(['test', 'test2']))}
    case CLEAR_HASHTAGS:
      return {...state, mentionResults: fromJS([])}
    default:
      return state
  }
}

// Selectors

export const getMentionResults = createSelector(
  (state) => state.hyloEditor,
  (state, props) => state.mentionResults
)

export const getHashtagResults = createSelector(
  (state) => state.hyloEditor,
  (state, props) => state.hasTagResults
)
