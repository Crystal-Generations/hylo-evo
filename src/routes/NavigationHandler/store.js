const NAVIGATE = 'NAVIGATE'
const RESET_NAVIGATION = 'RESET_NAVIGATION'

export function navigate (to) {
  return {
    type: NAVIGATE,
    payload: {to}
  }
}

export function resetNavigation (to) {
  return {type: RESET_NAVIGATION}
}

const reducer = (state = null, { type, error, payload, meta }) => {
  switch (type) {
    case NAVIGATE:
      return payload.to
    case RESET_NAVIGATION:
      return null
  }
  return state
}

export default reducer
