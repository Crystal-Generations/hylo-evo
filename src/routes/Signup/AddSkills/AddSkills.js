import React, { Component } from 'react'
import { get } from 'lodash/fp'
import { isEmpty } from 'lodash'
import SignupModalFooter from '../SignupModalFooter'
import LeftSidebar from '../LeftSidebar'

import '../Signup.scss'

export default class AddSkills extends Component {
  clickHandler = (skill) => {
    this.props.addSkill(skill.name)
  }

  getRemainingSkills = () => {
    const addedSkills = []
    for (let index in this.props.skills) {
      const skill = this.props.skills[index]
      addedSkills.push(skill.name)
    }
    return skills.filter(function (skill) {
      return (addedSkills.indexOf(skill.name) === -1)
    })
  }

  submit = () => {
    this.props.goToNextStep()
  }

  previous = () => {
    this.props.goToPreviousStep()
  }

  componentWillMount = () => {
    const { currentUser } = this.props
    if (get('settings.signupInProgress', currentUser) === 'false') this.props.goBack()
  }

  componentDidMount = () => {
    this.props.fetchMySkills()
  }
  render () {
    return <div styleName='flex-wrapper'>
      <LeftSidebar
        header='Share your unique super powers!'
        body="What skills are you known for? The more skills you add, the more relevant the content. It's like magic."
      />
      <div styleName='right-panel'>
        <span styleName='white-text step-count'>STEP 3/4</span>
        <br />
        <div styleName='center'>
          <input
            styleName='signup-input center-text signup-padding large-input-text gray-bottom-border'
            autoFocus
            onChange={this.handleInputChange}
            placeholder={'How can you help?'}
            readOnly
          />
        </div>
        <div>
          {skills && <div styleName='skills'>
            {this.getRemainingSkills().map((skill, index) =>
              <Pill key={index} skill={skill} handler={() => this.clickHandler(skill)} handlerArg={'name'} />
            )}
          </div>}
        </div>
        <div>
          { !isEmpty(this.props.skills) && <div styleName='your-skills'>
            <div styleName='your-skills-label'><span>Your Skills:</span></div>
            <div styleName='your-skills-list'>
              {this.props.skills.map((skill, index) =>
                <Pill key={index} skill={skill} handler={() => this.clickHandler(skill)} handlerArg={'name'} />
              )}
            </div>
          </div>}
        </div>
        <div>
          <SignupModalFooter previous={this.previous} submit={this.submit} continueText={'Boom, Done'} />
        </div>
      </div>
    </div>
  }
}

export function Pill ({skill, handler, handlerArg}) {
  return <span styleName='skill' onClick={() => handler(skill[handlerArg])}>
    {skill.name}
  </span>
}

const skills = [
  {name: 'Writing'},
  {name: 'Design'},
  {name: 'Project Management'},
  {name: 'Photography'},
  {name: 'Facilitation'},
  {name: 'Media'},
  {name: 'Community Organizing'},
  {name: 'Technology'},
  {name: 'Social Media'},
  {name: 'Event Planning'},
  {name: 'Education'},
  {name: 'Communications'}
]
