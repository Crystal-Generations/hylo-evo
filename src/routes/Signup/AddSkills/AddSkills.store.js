import { get } from 'lodash/fp'

export const SET_AUTOCOMPLETE = `SET_AUTOCOMPLETE`
export const FETCH_SKILLS_AUTOCOMPLETE = `FETCH_SKILLS_AUTOCOMPLETE`

export function setAutocomplete (autocomplete) {
  return {
    type: SET_AUTOCOMPLETE,
    payload: autocomplete
  }
}

export function getAutocomplete (state) {
  return state.AddSkills.autocomplete
}

export function fetchSkills (autocomplete) {
  return {
    type: FETCH_SKILLS_AUTOCOMPLETE,
    graphql: {
      query: `query ($autocomplete: String) {
        skills (first: 10, autocomplete: $autocomplete) {
          items {
            id
            name
          }
        }
      }`,
      variables: {
        autocomplete
      }
    },
    meta: {
      extractModel: 'Skill',
      extractQueryResults: {
        getItems: get('payload.data.skills')
      }
    }
  }
}

// Reducer
const defaultState = {
  skills: []
}

export default function reducer (state = defaultState, action) {
  const { error, type, payload } = action
  if (error) return state

  switch (type) {
    case SET_AUTOCOMPLETE:
      return {
        ...state,
        autocomplete: payload
      }
    default:
      return state
  }
}
