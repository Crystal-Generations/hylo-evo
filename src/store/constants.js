export const _PENDING = '_PENDING'

export const CREATE_COMMENT = 'CREATE_COMMENT'
export const CREATE_COMMENT_PENDING = CREATE_COMMENT + _PENDING

export const FETCH_COMMENTS = 'FETCH_COMMENTS'
export const FETCH_FEED_ITEMS = 'FETCH_FEED_ITEMS'
export const FETCH_POST = 'FETCH_POST'
export const FETCH_POSTS = 'FETCH_POSTS'

export const FETCH_CURRENT_USER = 'FETCH_CURRENT_USER'
export const TOGGLE_COMMUNITIES_DRAWER = 'TOGGLE_COMMUNITIES_DRAWER'

export const EXTRACT_MODEL = 'EXTRACT_MODEL'
