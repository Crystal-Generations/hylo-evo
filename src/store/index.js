import React from 'react'
import { createStore, applyMiddleware, combineReducers, compose } from 'redux'
import { Provider } from 'react-redux'
import { ConnectedRouter, routerReducer, routerMiddleware } from 'react-router-redux'
import createHistory from 'history/createBrowserHistory'
import { compact, flatten } from 'lodash'
import reducers from '../reducers'
import routes from '../routes'

const history = createHistory()

// const middleware = compact(flatten([
//   routerMiddleware(history)
// ]))

const middleware = routerMiddleware(history)

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(
  combineReducers({
    // ...reducers,
    router: routerReducer
  }),
  composeEnhancers(
    applyMiddleware(middleware)
  )
)

const app = <Provider store={store}>
  <ConnectedRouter history={history}>
    {routes}
  </ConnectedRouter>
</Provider>

export default app
