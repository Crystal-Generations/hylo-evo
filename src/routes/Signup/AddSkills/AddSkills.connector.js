import { connect } from 'react-redux'
import { setAutocomplete, getAutocomplete, fetchSkills } from './AddSkills.store'

export function mapStateToProps (state, props) {
  const autocomplete = getAutocomplete(state)
  return {
    autocomplete
  }
}
export const mapDispatchToProps = {
  setAutocomplete,
  fetchSkills
}

export default connect(mapStateToProps, mapDispatchToProps)
