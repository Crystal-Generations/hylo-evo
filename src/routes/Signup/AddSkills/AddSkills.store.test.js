import { mapDispatchToProps } from './AddSkills.store'

describe('AddSkills.store', () => {
  it('should match latest snapshot', () => {
    expect(mapDispatchToProps).toMatchSnapshot()
  })
})
