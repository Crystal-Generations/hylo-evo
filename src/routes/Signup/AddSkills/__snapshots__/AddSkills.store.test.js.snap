// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSkills.store should match latest snapshot 1`] = `undefined`;
